doc/user/layouts/shortcodes/views-indexes/index-considerations.html

```diff
@@ -0,0 +1,23 @@
+Specific instances where indexes can be useful to improve performance include:
+
+- When used in ad-hoc queries.
+
+- When used by multiple queries within the same cluster.
+
+- When used to enable delta joins.
+
+For more information, see [Optimization](/transform-data/optimization).
+
+**However, before creating an index**, consider the following:
```

Comment:
Could also break this into two files.  That way, we could have
## Usage patterns

Before creating an index, ...

### Indexes on a view vs. mat views
{{table}}

### Indexes and query optimization

The first part of this file.